subsys/pm/policy.c

```diff
@@ -130,6 +139,20 @@ static void update_next_event(uint32_t cyc)
 	next_event_cyc = new_next_event_cyc;
 }
 
+#ifdef CONFIG_PM_LOW_LATENCY_EVENTS
+int32_t pm_policy_next_low_latency_event_ticks(void)
+{
+	int32_t cyc_evt = -1;
+
+	if ((next_low_latency_event) &&
+	   (next_low_latency_event->value_cyc > 0)) {
+		cyc_evt = next_low_latency_event->value_cyc - k_cycle_get_32();
+	}
+
+	return k_cyc_to_ticks_ceil32(cyc_evt);
```

Comment:
If there is no risk of rolling over the ticks value on this conversion, please comment why.

Reply:
@hubertmis if I'm not mistaken: for nRF54h this actually divides the `cyc_evt` by https://github.com/nrfconnect/sdk-zephyr/blob/20ab87f134cfd57a64747e101f4a5693b14c6a67/soc/nordic/Kconfig.defconfig#L24. For cases where RTC is used both SYS_CLOCK_HW_CYCLES_PER_SEC and SYS_CLOCK_TICKS_PER_SEC are qual so the value should be not changed. Correct me please if I'm wrong.

Reply:
I didn't check these details. However, this function is a generic function expected to work for all SoCs regardless of their HW_CYCLES_PER_SEC and the TICKS_PER_SEC configuration setting that can be modified for each application.

That's why I would like to be sure that this conversion is sound for all cases (when cycles are faster than ticks, when ticks are faster than cycles - if that's possible, if one is a multiplication of another, and if not). If some logic I'm missing here ensures this, I would appreciate a comment in the code explaining why.

Reply:
If ticks are faster than cycles (it seems to be hard achieve in reality, or doesn't make much sense, does it?) then it can overflow.

Reply:
If that's the case, something like this would clarify why this solution is sound:

```
/* Make sure the conversion below never overflows. */
static_assert(CONFIG_HW_CYCLES_PER_SEC >= CONFIG_TICKS_PER_SEC);
return k_cyc_to_ticks_ceil32(cyc_evt);
```

Reply:
@hubertmis and @ppryga-nordic What I know is that the k_cyc_to_ticks_ceil32 function is calling z_tmcvt_32 which is a generic function that will take into account if CONFIG_HW_CYCLES_PER_SEC is bigger/less than CONFIG_TICKS_PER_SEC
See its implementation here :
https://github.com/zephyrproject-rtos/zephyr/blob/cb17f5f5a4af168d535227a14923bd28a8c17ee6/include/zephyr/sys/time_units.h#L220
Please correct me if I am wrong here

Reply:
Wow, that's complicated. However, I'm still not convinced there is no risk of overflows. I don't understand this macro. Also, it seems the docs don't clearly state which conversions can roll over and which are overflow-free-guaranteed.

I'm asking these questions because I've seen many bugs around timer overflows that are hard to reproduce because they happen e.g. after 11 hours of a device operation or in some corner cases unfeasible to test (e.g. scheduling a low-latency-event in 71 minutes).